include/linux/soundwire/sdw_intel.h

```diff
@@ -110,6 +110,8 @@ struct sdw_intel_stream_free_data {
 	int link_id;
 };
 
+struct snd_pcm_substream;
+
```

Comment:
IIRC we don't actually need to pass the substream to the callback. I think we only need the direction?

Reply:
87c15c159bc6310aa098d1faef6ce013343bb4d5

 ASoC/soundwire: intel: simplify callbacks for params/hw_free
    
    We don't really need to pass a substream to the callback, we only need
    the direction. No functionality change, only simplification to enable
    improve suspend with paused streams.

Reply:
@plbossart, I honestly don't know, but the compile time warning bugged me when I rebased my working branch.

Reply:
yes, that was more a comment for @bardliao on the definition of trigger itself.

Reply:
@ujfalusi @plbossart Please see https://github.com/thesofproject/linux/pull/3507

Reply:
> @ujfalusi @plbossart Please see #3507

@bardliao, OK, let's pursue that route, closing this PR.